app/main.py

```diff
@@ -1,25 +1,106 @@
-from abc import ABC
+from abc import ABC, abstractmethod
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: Any) -> None:
+        self.name = name
+
+    def __get__(self, instance: Any, owner: Any) -> None:
+        return getattr(instance, self.name)
+
+    def __set__(self, instance: Any, value: Any) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value must be an integer")
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(
+                f"Value must be between "
+                f"{self.min_amount} and {self.max_amount}"
+            )
+        setattr(instance, self.name, value)
 
 
 class Visitor:
-   pass
+
+    def __init__(
+            self,
+            name: str,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age: IntegerRange,
+            weight: IntegerRange,
+            height: IntegerRange
+    ) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
+
+    @abstractmethod
+    def validate(self, visitor: object) -> None:
```

Comment:
Do not implement the `validate` method, remove it and rely on the descriptor class instead